cmd/charm/charmcmd/cmd_test.go

```diff
@@ -40,36 +45,45 @@ import (
 // its context directory set to dir. It returns the output of the command
 // and its exit code.
 func run(dir string, args ...string) (stdout, stderr string, exitCode int) {
-	return runWithInput(dir, "", args...)
-}
-
-// runWithInput runs a charm plugin subcommand with the given arguments,
-// its context directory set to dir, and with standard input set to in.
-// It returns the output of the command and its exit code.
-func runWithInput(dir string, in string, args ...string) (stdout, stderr string, exitCode int) {
 	// Remove the warning writer usually registered by cmd.Log.Start, so that
 	// it is possible to run multiple commands in the same test.
 	// We are not interested in possible errors here.
 	defer loggo.RemoveWriter("warning")
 	var stdoutBuf, stderrBuf bytes.Buffer
 	ctxt := &cmd.Context{
 		Dir:    dir,
-		Stdin:  strings.NewReader(in),
+		Stdin:  nil,
 		Stdout: &stdoutBuf,
 		Stderr: &stderrBuf,
 	}
 	exitCode = cmd.Main(charmcmd.New(), ctxt, args)
 	return stdoutBuf.String(), stderrBuf.String(), exitCode
 }
 
+type homeSuite struct{}
+
+func (s *homeSuite) Init(c *qt.C) {
+	oldHome := utils.Home()
+	c.AddCleanup(func() {
```

Comment:
For readability purposes, I think it might be nice to add the cleanup after the thing it's cleaning up so it reads like a defer statement (and elsewhere).